target/linux/ar71xx/patches-4.4/702-MIPS-ath79-ubi_mtd_to_cmdline_append.patch

```diff
@@ -0,0 +1,44 @@
+--- a/arch/mips/ath79/common.c
++++ b/arch/mips/ath79/common.c
+@@ -161,3 +161,20 @@
+ }
+ EXPORT_SYMBOL_GPL(ath79_flash_release);
+
++static int __append_ubi_mtd_to_cmdline(int act)
++{
++	static int need_append = 0;
++	if(act >= 0)
++		need_append = act;
++	return need_append;
++}
++
++void append_ubi_mtd_to_cmdline(void)
++{
++	__append_ubi_mtd_to_cmdline(1);
++}
++
++int is_ubi_mtd_to_cmdline_need_append(void)
++{
++	return __append_ubi_mtd_to_cmdline(-1);
++}
+--- a/arch/mips/ath79/common.h
++++ b/arch/mips/ath79/common.h
+@@ -31,5 +31,7 @@
+ void ath79_gpio_output_select(unsigned gpio, u8 val);
+ int ath79_gpio_direction_select(unsigned gpio, bool oe);
+ void ath79_gpio_init(void);
++void append_ubi_mtd_to_cmdline(void);
++int is_ubi_mtd_to_cmdline_need_append(void);
+
+ #endif /* __ATH79_COMMON_H */
+--- a/arch/mips/ath79/prom.c
++++ b/arch/mips/ath79/prom.c
+@@ -145,6 +145,8 @@
+	    strstr(arcs_cmdline, "board=711Gr100") ||
+	    strstr(arcs_cmdline, "board=922gs"))
+		ath79_prom_append_cmdline("console", "ttyS0,115200");
++	if(is_ubi_mtd_to_cmdline_need_append())
++		ath79_prom_append_cmdline("ubi.mtd", "ubi");
```

Comment:
You were right. Auto-attaches UBI partitions called "ubi" works ok. I dont know why but in the previous try I couldnt boot with ubi auto-attaches but now (with the current trunk), everything works fine. I tested it on two different devices (rb751 and rb433) and all ok.